### DISCUSSION TITLE
Solved: It takes a long time to start the shell on Mac

### DISCUSSION THREAD

**Discussion by catfist** [Taitava/obsidian-shellcommands#234] [Debugging and testing]
I'm spending a few seconds on commands that simply `echo foo`.
On the other hand, if I execute `/bin/bash -x` with iterm, or cmd/PowerShell, it complete instantly.
If I write `echo "source ~/.bashrc` in first line of `~/.bashrc`, it also not noticed instantly. In other words, it isn't time to process `~/.bashrc`.
This is the same for bash and zsh.

**Comment by Taitava** [comment]
> I'm spending a few seconds on commands that simply `echo foo`.
> On the other hand, if I execute `/bin/bash -x` with iterm, or cmd/PowerShell, it complete instantly.

Hmm, I'm not sure if I'm able to provide any solution to this, but let's still inspect. Would be nice to know why there is a delay. Can you please try to use [Templater](https://github.com/SilentVoid13/Templater) to test your Bash command? [Here is another discussion with instructions on how to do it](https://github.com/Taitava/obsidian-shellcommands/discussions/187#discussioncomment-2431781). The idea is to see if the lag is caused by SC or if it's just something that happens for you whenever Bash commands are executed from Obsidian.

> If I write `echo "source ~/.bashrc` in first line of `~/.bashrc`, it also not noticed instantly. In other words, it isn't time to process `~/.bashrc`.
> This is the same for bash and zsh.

I'd guess `~/.bashrc` is never used when executing shell commands via an Electron app (such as Obsidian), but this is just my guess. I think this is a different problem and not related to the delay problem. I'm not happy with the fact that profile files are not invoked when using the _Shell commands_ plugin. I have been thinking on a robust and clean way how to get those profile files included automatically, but haven't come up with any plan yet. So far you could try to prefix your shell commands with `source ~/.bashrc`.

**Reply by Taitava** [reply]
Also, another discussion mentions that reinstalling Obsidian with newest version has solved a delay problem. https://github.com/Taitava/obsidian-shellcommands/discussions/227

**Reply by catfist** [reply]
I tried execution from Templater, but failed resolve error.
So I reinstalled Obsidian according to #227 , and confirmed the resolution.
Thanks for advice🙂

**Reply by Taitava** [reply]
Nice to know that the delay has now gone away! 🙂

**Comment by Taitava** [comment]
# Documentation
Just for the record, I created [a documentation page for delay problems](https://publish.obsidian.md/shellcommands/Problems/Delay+when+executing+a+shell+command). It doesn't contain any new information, but perhaps makes it easier to find the information in the future. 🙂
